components/wmtk_components/extract_subset/internal/extract_subset_2d.cpp

```diff
@@ -0,0 +1,130 @@
+#include "extract_subset_2d.hpp"
+
+wmtk::TriMesh extract_subset_2d(
+    const std::vector<Eigen::Vector2d>& points,
+    Eigen::MatrixXi& triangles, std::vector<size_t> tag){
+
+    int nb_vertex = points.size();
+    int nb_vertex_in = 0;
+    int nb_tri_in = tag.size();
+    // maintain a vector of bool, true if an old vertex is preserved after extraction
+    std::vector<bool> vertices_in_bool(nb_vertex);
+    for (int k = 0; k < nb_vertex; ++k) {vertices_in_bool[k] = false;}
+
+    Eigen::MatrixXi faces_in;
+    faces_in.resize(nb_tri_in, 3);
+
+    //tag the preserved ones and count number of vertex in new extraction
+    for (size_t k = 0; k < nb_tri_in; ++k){
+        for (size_t k2 = 0; k2 < 3; ++k2) {
```

Comment:
I don't know why you are using `k` and `k2` here but the convention is to use `i` and `j`. Please change that, otherwise it might be confusing.